Implement Scheduler CLI state changer

bembelimen opened the pull request:
See #20

bembelimen committed:
Implement Scheduler CLI state changer

bembelimen committed:
Update TasksStateCommand.php

ditsuke commented on libraries/src/Console/TasksStateCommand.php:
```diff
@@ -0,0 +1,182 @@
+	 *
+	 * @since __DEPLOY_VERSION__
+	 */
+	const STATE_PUBLISH = 1;
```
```suggestion
	const STATE_ENABLE = 1;
```

ditsuke commented on libraries/src/Console/TasksStateCommand.php:
```diff
@@ -0,0 +1,182 @@
+	 *
+	 * @since __DEPLOY_VERSION__
+	 */
+	const STATE_UNPUBLISH = 0;
```
```suggestion
	const STATE_DISABLE = 0;
```

ditsuke commented on libraries/src/Console/TasksStateCommand.php:
```diff
@@ -0,0 +1,182 @@
+
+		if (!is_numeric($state))
+		{
+			$state = ArrayHelper::getValue(['publish' => self::STATE_PUBLISH, 'unpublish' => self::STATE_UNPUBLISH,
```
```suggestion
			$state = ArrayHelper::getValue(['enable' => self::STATE_ENABLE, 'disable' => self::STATE_DISABLE,
```

ditsuke commented on libraries/src/Console/TasksStateCommand.php:
```diff
@@ -0,0 +1,182 @@
+			);
+		}
+
+		while (!strlen($state) || !in_array($state, [self::STATE_PUBLISH, self::STATE_UNPUBLISH, self::STATE_TRASH]))
```
```suggestion
		while (!strlen($state) || !in_array($state, [self::STATE_ENABLE, self::STATE_DISABLE, self::STATE_TRASH]))
```

ditsuke commented on libraries/src/Console/TasksStateCommand.php:
```diff
@@ -0,0 +1,182 @@
+
+			if (!is_numeric($state))
+			{
+				$state = ArrayHelper::getValue(['publish' => self::STATE_PUBLISH, 'unpublish' => self::STATE_UNPUBLISH,
```
```suggestion
				$state = ArrayHelper::getValue(['enable' => self::STATE_ENABLE, 'unpublish' => self::STATE_DISABLE,
```

ditsuke commented on libraries/src/Console/TasksStateCommand.php:
```diff
@@ -0,0 +1,182 @@
+
+		$table = $taskModel->getTable();
+
+		$action = array_search($state, ['publish' => self::STATE_PUBLISH, 'unpublish' => self::STATE_UNPUBLISH, 'trash' => self::STATE_TRASH]);
```
```suggestion
		$action = array_search($state, ['enable' => self::STATE_ENABLE, 'disable' => self::STATE_DISABLE, 'trash' => self::STATE_TRASH]);
```

ditsuke commented on libraries/src/Console/TasksStateCommand.php:
```diff
@@ -0,0 +1,182 @@
+
+		while (!strlen($state) || !in_array($state, [self::STATE_PUBLISH, self::STATE_UNPUBLISH, self::STATE_TRASH]))
+		{
+			$state = (string) $this->ioStyle->ask('Should the state be "publish" (1), "unpublish" (0) or "trash" (-2)');
```
```suggestion
			$state = (string) $this->ioStyle->ask('Should the state be "enable" (1), "disable" (0) or "trash" (-2)');
```

ditsuke commented on libraries/src/Console/TasksStateCommand.php:
```diff
@@ -0,0 +1,182 @@
+	protected function configure(): void
+	{
+		$this->addOption('id', 'i', InputOption::VALUE_REQUIRED, 'The id of the task to change');
+		$this->addOption('state', 's', InputOption::VALUE_REQUIRED, 'Set the new state of the task, can be 1/publish, 0/unpublish, -2/trash.');
```
```suggestion
		$this->addOption('state', 's', InputOption::VALUE_REQUIRED, 'Set the new state of the task, can be 1/enable, 0/disable, -2/trash.');
```

ditsuke commented on libraries/src/Console/TasksStateCommand.php:
```diff
@@ -0,0 +1,182 @@
+		$help = "<info>%command.name%</info> changes the state of a task.
+		\nUsage: <info>php %command.full_name%</info>";
+
+		$this->setDescription('Publish/Unpublish/Trash a task');
```
```suggestion
		$this->setDescription('Enable/Disable/Trash a task');
```

ditsuke committed:
update publish/unpublish => enable/disable

ditsuke committed:
Add check for checked out tasks

Adds check for checked out state of tasks, so it doesn't silently fail
anymore if such a task is attempted to be updated.

ditsuke committed:
Improve static analysis support.

Adds type hints for classes so static analysis and IDE autocomplete
works better.

ditsuke committed:
Update exit codes and fix typos

- Updates failure exit codes so all are now distinct.
- Fixes typos and other minor things.
- Uses syntactic string composition instead of concatenation.

ditsuke commented:
thanks!
